tests/unit/s2n_rsa_pss_rsae_test.c

```diff
@@ -241,35 +241,49 @@ int main(int argc, char **argv)
         /* Test: If they share the same RSA key,
          * an RSA cert and an RSA_PSS cert are equivalent for PSS signatures. */
         {
-            struct s2n_pkey rsa_pss_public_key;
```

Comment:
What are the changes you're making to this test? How is this related to the bug?

Reply:
The change is related to [this comment](https://github.com/aws/s2n-tls/pull/4574/files#diff-6ddd27035d4765c4a56db3d1d8696325a2fd61dc7b36925d601adfca71d4eb82R258-R260). 

This test works by creating s2n_pkeys from reading RSA and RSA_PSS certificates, and then setting the EVP pkeys to be the same. To do this, `EVP_PKEY_set1_RSA()` is used, which overrides the EVP pkey type to RSA (even though it was originally an RSA_PSS key). Because of the new check, the pkey type needs to match the signature algorithm, so it's not possible to test signing RSA_PSS with the overridden pkey type.

To fix this I reversed which pkey is overridden. So it now overrides the RSA key with the RSA_PSS key, which allows the RSA_PSS key to continue having its RSA_PSS type.